scripts/irods/test/rule_texts_for_tests.py

```diff
@@ -606,6 +606,25 @@
 INPUT null
 OUTPUT ruleExecOut
 '''
+rule_texts['irods_rule_engine_plugin-irods_rule_language']['Test_Rulebase']['test_msiExecCmd_closeAllL1Desc__issue_6623_rule'] ='''
+execute_hello {{
+    msiDataObjOpen("{logical_path}", *FD);
+    msiExecCmd('hello', '', '', '', '', *out);
+    msiDataObjClose(*FD, *Status);
+}}
+
+INPUT null
+OUTPUT ruleExecOut
+'''
+rule_texts['irods_rule_engine_plugin-irods_rule_language']['Test_Rulebase']['test_msiExecCmd_closeAllL1Desc__issue_6623_pep'] ='''
+pep_resource_close_post(*inst, *ctx, *out) {{
+    msiGetSystemTime(*time_str, '');
+    msiAddKeyVal(*key_val_pair,"{attr}","*time_str");
+    msiAssociateKeyValuePairsToObj(*key_val_pair,"{logical_path}","-d");
+    # Sleep to ensure that the actual close doesn't occur in the same second
+    msiSleep("1", "0");
+}}
+'''
```

Comment:
Consider moving these into their respsective test.

Having part of the test logic in a separate file introduces more hurdles to understanding the test.

If you see benefits to continuing to use this file, I'd be interested in hearing them.

Reply:
I created #6677 to remove this file. Might as well just go ahead and do it. ;p